src/contracts/Collection/AbstractInMemoryCollection.php

```diff
@@ -52,15 +52,15 @@ public function count(): int
     }
 
     /**
-     * @return static<TValue>
+     * @return self
```

Comment:
`static` (with type as well) is used to indicate to static analysis that this method returns an object of the same type as the method is called upon.

If it really is necessary, then this phpdoc should be replaced with one specific to our static analysis tool.

```suggestion
     * @phpstan-return static<TValue>
```

Reply:
I'll propose the PHPStan one